### PR TITLE
Add io.github.mki1967.mki3dgame

### PR DESCRIPTION
**Pull request by mki1967**
A simple game of searching in stages designed with [mki3d](https://mki1967.github.io/mki3d/).
It has been written in Go, using [go-gl](https://github.com/go-gl) packages.

### CODE REVIEW

File: io.github.mki1967.mki3dgame.json [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "1.6",
+    "sdk": "org.freedesktop.Sdk",
+    "sdk-extensions": [
+        "org.freedesktop.Sdk.Extension.golang"
+    ],
+    "command": "mki3dgame",
+    "finish-args": ["--share=ipc", "--socket=x11", "--socket=pulseaudio", "--share=network", "--device=dri", "--socket=wayland"],
+    "cleanup-commands": [
+	"rm -rf /app/go # remove working space"
```
Comment (TingPing) [inline io.github.mki1967.mki3dgame.json]: `"cleanup": [ "/go"` is a bit cleaner.

File: io.github.mki1967.mki3dgame.json [hunk 2]
```diff
@@ -0,0 +1,108 @@
+{
+    "app-id": "io.github.mki1967.mki3dgame",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "1.6",
+    "sdk": "org.freedesktop.Sdk",
+    "sdk-extensions": [
+        "org.freedesktop.Sdk.Extension.golang"
+    ],
+    "command": "mki3dgame",
+    "finish-args": ["--share=ipc", "--socket=x11", "--socket=pulseaudio", "--share=network", "--device=dri", "--socket=wayland"],
```
Comment (TingPing) [inline io.github.mki1967.mki3dgame.json]: Put these on their own line please.

File: io.github.mki1967.mki3dgame.json [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    "cleanup-commands": [
+	"rm -rf /app/go # remove working space"
+    ],
+    "modules": [
+	{
+	    "name": "io.github.mki1967.mki3dgame",
+	    "only-arches": [
```
Comment (TingPing) [inline io.github.mki1967.mki3dgame.json]: This can be removed since the `flathub.json` file prevents it building anywhere else.

**Comment by TingPing** [comment]
bot, build io.github.mki1967.mki3dgame

**Comment by TingPing** [comment]
In the appdata file:

```xml
  <name>io.github.mki1967.mki3dgame</name>
```

The name should be human readable.

Adding `<releases>` would be nice too.

**Comment by TingPing** [comment]
In the desktop file `Terminal=true`, if it is a graphical application why is that set?

**Comment by mki1967** [comment]
`Terminal=true` is set, since the  terminal is used to display text messages to the user. The initial help screen, for example.

**Comment by nedrichards** [comment]
bot, build io.github.mki1967.mki3dgame

**Comment by eszlari** [comment]
> Terminal=true is set, since the terminal is used to display text messages to the user. The initial help screen, for example.

Couldn't you use `zenity` for that?

**Comment by mki1967** [comment]
@eszlari  Thank you for the suggestion. I will have to learn `zenity`. Do you know any Go packages for `zenity` or should I rather use [spawning processes](https://gobyexample.com/spawning-processes) ?

**Comment by eszlari** [comment]
There are Go bindings for gtk:
https://github.com/gotk3/gotk3

Zenity is like a command line wrapper for gtk. It is commonly used in shell scripts. So yes,  it is meant to be used as a dedicated process.

**Comment by mki1967** [comment]
I have made the changes for `zenity`. I hope it is all right this time.

**Comment by nedrichards** [comment]
bot, build io.github.mki1967.mki3dgame

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/io.github.mki1967.mki3dgame

**Pull request closed without merging**